### DISCUSSION TITLE
Questions about first installation with Raspberry Pi Pico (Rp2040)

### DISCUSSION THREAD

**Discussion by szicx** [awawa-dev/HyperHDR#660] [Q&A]
Hello, I have a few questions for my first installation of **HyperHDR**.

_Here is the list of components I want to use:_

HDMI Splitter EZ-SP12H2 : https://www.amazon.fr/Splitter-HDMI-Mini-60Hz-synchronisation/dp/B083JVPXDY/ref=sr_1_6?keywords=ezcoo&qid=1698273700&sr=8-6&th=1

HDMI USB Grabber MS2130 model B : https://fr.aliexpress.com/item/1005004990708725.html?gatewayAdapt=glo2fra

4M LED STRIP SK6812 RGBCW 60L/M: https://fr.aliexpress.com/item/32476317187.html?gatewayAdapt=glo2fra

Power supply : not yet determined

_HyperHDR runs on my Windows PC, I don't want to use Raspberry for the server._

To control the LEDs, I do not want to use an ESP because I am not comfortable with the idea of building the card for the 3.3 V to 5 V level selector. I would like to use the solution proposed in the guide, with a **Raspberry Pi Pico (Rp2040) which has an integrated 3.3 V to 5 V level selector** with **HyperSerialPico**.

As I want to use **STRIP SK6812 RGBCW LEDs**, I don't know if I should choose **Pimoroni Plasma Stick 2040 W** or **Pimoroni Plasma 2040**?

One has 1 output, the second have 2 output.

Pimoroni Plasma Stick 2040 W: https://shop.pimoroni.com/products/plasma-stick-2040-w?variant=40359072301139
Pimoroni Plasma 2040: https://shop.pimoroni.com/products/plasma-2040?variant=39410354847827

My television is 55" and I'm going to use 4 meters of SK6812 RGBCW.

Thank you for your valuable help, I suppose my questions may be stupid for experts but I am discovering HyperHDR, and I am impressed by the work you have done, congratulations!

**Have a good day readers!**

PS: I currently have HyperHDR V19, I see that I need to use HyperHDR V20 Beta for HyperSerialPico support, where can I find this version?

**Comment by dbell669** [comment]
I would go with the non W version. The only advantage it has over the regular one is that its wireless according to 

https://core-electronics.com.au/guides/raspberry-pi-pico-w-vs-pico-whats-the-difference/

The dual outputs on the regular 2040 mean that you can have the multi segment setup for a higher refresh rate. Even if you are only going to have about 240 Leds in total it means it would be easier to scale the system up in the future for a ~2USD Difference.

**Reply by szicx** [reply]
Hello, thank you for your answer.

So you advise me to go with this model? https://shop.pimoroni.com/products/plasma-2040?variant=39410354847827

Small additional question, do I have to cut my LEDS strip into several sections? Is it not possible to leave the LED strip in one piece for the SK6812?

Thanks

**Reply by dbell669** [reply]
Thats the correct 2040. As for your sectioning question, it depends on a few factors. 

The first being the way you are going to place the lights themselves. if you look at the "Multi Segment Wiring" Section of [https://github.com/awawa-dev/HyperSerialESP32](url) you can see that its broken up into 2 sets of 3 sections. (The 2 data lines starting from the ESP32 or Pico in your case) This is to compensate for the 90 degree angles for the TV Corners.  I would suggest looking at Section H of [https://www.hyperhdr.eu/2023/02/ultimate-guide-on-how-to-build-led.html#chapter3](url) in order to see what I'm talking about in a real life situation instead of a diagram. Having it sit at a 45 degree angle gives you a better projection of the LED light. You could have the lights perpendicular to the tv in order to have only 2 sections, but when I attempted that it only worked as well as it did because my Tv was less than like 3.8 CM (1.5 In) from the wall.

The second reason is the refresh rate.. Both of the previous links have sections concerning the benefits parallel segment deployment. Having the Level shifter built into the pico removes a lot of the complexity in setting that up.

**Reply by szicx** [reply]
Ok I understand thank you, the developer enlightened me on this I just have to validate the connection diagram.
Have good day!

**Comment by awawa-dev** [comment]
Hi
Since the Plasma Stick 2040 W is actually two PCBs glued one on top of the other, this model basically has physically blocked GPIO outputs except the LED output. So you won't use it, for example, for HyperSPI, if RP2040 support will be added in the near future, or for other purposes: it can only drive Neopixel LEDs using single data lane. So it supports only single LED segment, although it doesn't matter in your setup because ~240 LEDs do not require the use of a multi-segment (recommended for higher than ~300 RGBW LEDs number).
Latest v20 beta installers can be found in the latest Github Action tab as artifacts (build from the 'master' branch).

**Reply by szicx** [reply]
Hello @awawa-dev , thank you for your answer.

So if I understand correctly, I can use the following configuration:

_HDMI Splitter EZ-SP12H2
HDMI USB Grabber MS2130 model B
4M LED STRIP SK6812 RGBCW 60L/M
Pimoroni Plasma Stick 2040 W:_ https://shop.pimoroni.com/products/plasma-stick-2040-w?variant=40359072301139

I flash **HyperSerialPico** on the **Plasma Stick 2040 W** like that:
Firmware **"Pimoroni_Plasma_Stick_2040_W.zip"** -> **"HyperSerialPico_sk6812Cold.uf2"**

With this configuration, i can use my LED strip in a single piece, no need for cutting.

I made a diagram for the electrical supply of the **Plasma Stick 2040 W** with the **LEDs**, is that correct? (I'm not sure about the 5v)

Thanks again, I'm trying to get as much information as possible before launching the project. Good day to you!

**Reply by awawa-dev** [reply]
Under no circumstances can you connect +5v from the LED power supply to Pico which is already powered via USB +5V line. Please edit your post and correct this diagram or delete it so that it does not suggest this solution to other users who may find it here 😉 
As you can read on the wiki or in a recent blog post, multi-segment or single-segment does not just mean cutting an LED strip, but solutions consist of smaller LED strips connected in one common direction or two opposite directions. A single-segment can be cut and composed of several smaller ones, as long as the direction of the signal line matches. As for the rest of the equipment, it is an individual choice, but one of many that is used.

**Reply by szicx** [reply]
I just deleted the previous diagram. So should it be like this? really sorry for the misunderstanding

![Sans titre-24](https://github.com/awawa-dev/HyperHDR/assets/149027458/d4e73c38-d3e3-4718-b646-f3d2703a0db3)

**Reply by awawa-dev** [reply]
No problem, it's better to ask first than to regret later that something is broken :) Now the diagram is OK.

**Reply by szicx** [reply]
Perfect thank. I will place an order for all the necessary equipment.

I will now try to understand how the single segment works. In the guide, I found this diagram for Single-segment, is it correct for my configuration?

BIG thanks for taking the time to inform me, I appreciate it.

![Screenshot_1](https://github.com/awawa-dev/HyperHDR/assets/149027458/23432e49-b89e-4edd-8ec0-b3e8ddfeb032)

**Reply by awawa-dev** [reply]
Yes, it's an example of single segment setup. Theoretically, you don't have to cut the LED strip at the corners for single segment and use one continuous strip, but: if you bend the LED strip, you reduce the conductivity of the copper paths on it,  also the corners may peel off and nearby LEDs may be at a different angle to the surface than the rest... easier to do with a 30LEDs/meter strip and using a frame with a 45 angle than on the flat surface of the back wall of the TV, I haven't tested it myself.

**Reply by szicx** [reply]
I will cut in the corners, cleaner to install. I will also power the LEDs from the beginning and the end like the diagram.

Is cutting in straight lines obligatory? if yes all how many LEDs? This is a point that I misunderstood, I would have liked not to have to cut the headband (except in the 4 corners)

Good evening!

**Reply by awawa-dev** [reply]
So only cut it at the corners, in the latest diagram these are 5 smaller LED strips: top, left, right and 2 on the bottom. I think you are confusing the cutting symbol in this diagram with the continuity symbol.

**Reply by szicx** [reply]
Ahhhh yes my bad, there are 5 pieces on this diagram. You are right, I was confused, I thought that the 2 black bars on LEDs on the diagram corresponded to cuts, but no.

Time to sleep in think ahah, thanks for all. Everything is clear now!

**Reply by Bambisky** [reply]
hi !
Sorry to ask that here, but i'm relatively new on github, I also wanna use the plasma 2040 but cannot find where to download the V20 beta sd image for a rpi4.
tia !

**Reply by awawa-dev** [reply]
Hi
The SD image for v20beta is not ready yet. However, you can install the Raspberry Pi OS (e.g. aarch64, lite version, latest is Bookworm edition) using the official Raspberry Pi Imager utility and then install v20beta via ssh **(make sure you enabled it when flashing the image!)**: read more here https://github.com/awawa-dev/HyperHDR/discussions/561#discussioncomment-7532808
